Update/upstream

jbearer opened the pull request.

demon-xxi committed:
txmgr: Add Value field support in Tx

demon-xxi committed:
Add txmgr constructor with Config

tynes committed:
op-chain-ops: wip

tynes committed:
builder: more working

tynes committed:
op-chain-ops: bugfixes

ajsutton committed:
op-challenger: Don't progress completed games

ajsutton committed:
op-challenger: Remove unused players from memory.

hamdiallam committed:
remove u256 type

OptimismBot committed:
Merge pull request #6891 from ethereum-optimism/changeset-release/develop

Version Packages

refcell committed:
Expose a cleanup function on the game player.

protolambda committed:
op-wheel: add engine-api utils

refcell committed:
Touches up the responder tests.

refcell committed:
Only resolve when dispute game won

dependabot[bot] committed:
build(deps-dev): bump @wagmi/core from 1.3.8 to 1.3.9

Bumps [@wagmi/core](https://github.com/wagmi-dev/wagmi/tree/HEAD/packages/core) from 1.3.8 to 1.3.9.
- [Release notes](https://github.com/wagmi-dev/wagmi/releases)
- [Changelog](https://github.com/wagmi-dev/wagmi/blob/main/packages/core/CHANGELOG.md)
- [Commits](https://github.com/wagmi-dev/wagmi/commits/@wagmi/core@1.3.9/packages/core)

---
updated-dependencies:
- dependency-name: "@wagmi/core"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump eslint-plugin-prettier from 4.0.0 to 4.2.1

Bumps [eslint-plugin-prettier](https://github.com/prettier/eslint-plugin-prettier) from 4.0.0 to 4.2.1.
- [Release notes](https://github.com/prettier/eslint-plugin-prettier/releases)
- [Changelog](https://github.com/prettier/eslint-plugin-prettier/blob/master/CHANGELOG.md)
- [Commits](https://github.com/prettier/eslint-plugin-prettier/compare/v4.0.0...v4.2.1)

---
updated-dependencies:
- dependency-name: eslint-plugin-prettier
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/eslint-plugin

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 5.60.1 to 5.62.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.62.0/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump nx from 16.7.2 to 16.7.3

Bumps [nx](https://github.com/nrwl/nx/tree/HEAD/packages/nx) from 16.7.2 to 16.7.3.
- [Release notes](https://github.com/nrwl/nx/releases)
- [Commits](https://github.com/nrwl/nx/commits/16.7.3/packages/nx)

---
updated-dependencies:
- dependency-name: nx
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

hamdiallam committed:
uint64 in tests

OptimismBot committed:
Merge pull request #6950 from ethereum-optimism/indexer.u256

fix(indexer) remove database.U256

mergify[bot] committed:
Merge branch 'develop' into refcell/remove-old-cannon-dirs

OptimismBot committed:
Merge pull request #6951 from ethereum-optimism/refcell/remove-old-cannon-dirs

feat(op-challenger): Game Player and Cannon Cleanup

mergify[bot] committed:
Merge branch 'develop' into dependabot/npm_and_yarn/nx-16.7.3

ajsutton committed:
op-challenger: Cache the completed status of the game

OptimismBot committed:
Merge pull request #6963 from ethereum-optimism/dependabot/npm_and_yarn/nx-16.7.3

build(deps-dev): bump nx from 16.7.2 to 16.7.3

mergify[bot] committed:
Merge branch 'develop' into dependabot/npm_and_yarn/wagmi/core-1.3.9

OptimismBot committed:
Merge pull request #6959 from ethereum-optimism/dependabot/npm_and_yarn/wagmi/core-1.3.9

build(deps-dev): bump @wagmi/core from 1.3.8 to 1.3.9

mergify[bot] committed:
Merge branch 'develop' into refcell/recent-game-loading

OptimismBot committed:
Merge pull request #6927 from ethereum-optimism/refcell/recent-game-loading

feat(op-challenger): Only Load Recent Games

mergify[bot] committed:
Merge branch 'develop' into aj/skip-complete-games

ajsutton committed:
op-challenger: Delete game files when complete

OptimismBot committed:
Merge pull request #6948 from ethereum-optimism/aj/skip-complete-games

op-challenger: Don't progress completed games

mergify[bot] committed:
Merge branch 'develop' into refcell/touchup-responder-tests

OptimismBot committed:
Merge pull request #6953 from ethereum-optimism/refcell/touchup-responder-tests

fix(op-challenger): Touchup Responder Test Suite

mergify[bot] committed:
Merge branch 'develop' into refcell/call-resolve

OptimismBot committed:
Merge pull request #6956 from ethereum-optimism/refcell/call-resolve

fix(op-challenger): Only Resolve Won Dispute Games

mergify[bot] committed:
Merge branch 'develop' into aj/delete-trace-files

ajsutton committed:
op-challenger: Make datadir required for all trace types

OptimismBot committed:
Merge pull request #6965 from ethereum-optimism/aj/delete-trace-files

op-challenger: Delete game files when complete

mergify[bot] committed:
Merge branch 'develop' into dependabot/npm_and_yarn/typescript-eslint/eslint-plugin-5.62.0

ajsutton committed:
op-challenger: Use different directories when testing proof generation

OptimismBot committed:
Merge pull request #6961 from ethereum-optimism/dependabot/npm_and_yarn/typescript-eslint/eslint-plugin-5.62.0

build(deps-dev): bump @typescript-eslint/eslint-plugin from 5.60.1 to 5.62.0

mergify[bot] committed:
Merge branch 'develop' into dependabot/npm_and_yarn/eslint-plugin-prettier-4.2.1

OptimismBot committed:
Merge pull request #6960 from ethereum-optimism/dependabot/npm_and_yarn/eslint-plugin-prettier-4.2.1

build(deps-dev): bump eslint-plugin-prettier from 4.0.0 to 4.2.1

mergify[bot] committed:
Merge branch 'develop' into aj/rename-data-dir

dependabot[bot] committed:
build(deps-dev): bump @testing-library/jest-dom from 5.17.0 to 6.0.1

Bumps [@testing-library/jest-dom](https://github.com/testing-library/jest-dom) from 5.17.0 to 6.0.1.
- [Release notes](https://github.com/testing-library/jest-dom/releases)
- [Changelog](https://github.com/testing-library/jest-dom/blob/main/CHANGELOG.md)
- [Commits](https://github.com/testing-library/jest-dom/compare/v5.17.0...v6.0.1)

---
updated-dependencies:
- dependency-name: "@testing-library/jest-dom"
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

OptimismBot committed:
Merge pull request #6966 from ethereum-optimism/aj/rename-data-dir

op-challenger: Make datadir required for all trace types

mergify[bot] committed:
Merge branch 'develop' into dependabot/npm_and_yarn/testing-library/jest-dom-6.0.1

OptimismBot committed:
Merge pull request #6954 from ethereum-optimism/dependabot/npm_and_yarn/testing-library/jest-dom-6.0.1

build(deps-dev): bump @testing-library/jest-dom from 5.17.0 to 6.0.1

dependabot[bot] committed:
build(deps-dev): bump @types/node from 12.20.20 to 20.5.3

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 12.20.20 to 20.5.3.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump lint-staged from 11.0.0 to 14.0.1

Bumps [lint-staged](https://github.com/okonet/lint-staged) from 11.0.0 to 14.0.1.
- [Release notes](https://github.com/okonet/lint-staged/releases)
- [Commits](https://github.com/okonet/lint-staged/compare/v11.0.0...v14.0.1)

---
updated-dependencies:
- dependency-name: lint-staged
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

tynes committed:
op-chain-ops: cleanup

OptimismBot committed:
Merge pull request #6962 from ethereum-optimism/dependabot/npm_and_yarn/lint-staged-14.0.1

build(deps-dev): bump lint-staged from 11.0.0 to 14.0.1

tynes committed:
op-chain-ops: validation + test coverage

tynes committed:
op-chain-ops: fix build

ajsutton committed:
op-e2e: Assert deletion of game data in fault proof e2e tests

tynes committed:
op-chain-ops: cleanup

tynes committed:
op-chain-ops: rename files

tynes committed:
lint: fix

mergify[bot] committed:
Merge branch 'develop' into dependabot/npm_and_yarn/types/node-20.5.3

OptimismBot committed:
Merge pull request #6955 from ethereum-optimism/dependabot/npm_and_yarn/types/node-20.5.3

build(deps-dev): bump @types/node from 12.20.20 to 20.5.3

mergify[bot] committed:
Merge branch 'develop' into aj/delete-files-e2e

OptimismBot committed:
Merge pull request #6970 from ethereum-optimism/aj/delete-files-e2e

op-e2e: Assert deletion of game data in fault proof e2e tests

trajan0x committed:
deprecate docker-compose

OptimismBot committed:
Merge pull request #6972 from trajan0x/fix/docker-compose-deprecation

make usage of docker compose v2 explicit

ajsutton committed:
Update to golang 1.20.7 and alpine 3.18 (#6969)

* Update to golang 1.20.7 and alpine 3.18

* Update go version in go.mod files.

* Use crypto/rand.Read instead of math/read.Read

* Handle error response

hamdiallam committed:
cascade delete and unique constraints

hamdiallam committed:
spacing

refcell committed:
Modify the monitor component to only progress games on new L1 block.

ajsutton committed:
Manage game data directories at the multi-game level (#6968)

* op-challenger: Manage allocation of game data directories in multi-game code.

* op-challenger: Delete all directories except required games

mergify[bot] committed:
Merge branch 'develop' into indexer.sql.constraints

OptimismBot committed:
Merge pull request #6973 from ethereum-optimism/indexer.sql.constraints

fix(indexer):Schema unique constraints and cascading deletes

refcell committed:
Fixes small comments in the disk manager following pr #6968.

hamdiallam committed:
merge cli/cmd

hamdiallam committed:
remote git version

hamdiallam committed:
fix api_test

hamdiallam committed:
update gitignore so cli.go is not ignored

hamdiallam committed:
remove indexer binary

refcell committed:
Fix datadir check missing game prefix

OptimismBot committed:
Merge pull request #6981 from ethereum-optimism/refcell/fix-missing-game-prefix

fix(op-e2e): Challenger Datadir Check

mergify[bot] committed:
Merge branch 'develop' into indexer.cli

OptimismBot committed:
Merge pull request #6978 from ethereum-optimism/indexer.cli

fix(indexer): merge cli/cmd packages. interrupt handling for api server

mergify[bot] committed:
Merge branch 'develop' into refcell/nit-fix-game-dir-comments

dependabot[bot] committed:
build(deps-dev): bump markdownlint from 0.24.0 to 0.30.0

Bumps [markdownlint](https://github.com/DavidAnson/markdownlint) from 0.24.0 to 0.30.0.
- [Changelog](https://github.com/DavidAnson/markdownlint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/DavidAnson/markdownlint/compare/v0.24.0...v0.30.0)

---
updated-dependencies:
- dependency-name: markdownlint
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @codechecks/client from 0.1.11 to 0.1.12

Bumps [@codechecks/client](https://github.com/codechecks/monorepo) from 0.1.11 to 0.1.12.
- [Commits](https://github.com/codechecks/monorepo/commits)

---
updated-dependencies:
- dependency-name: "@codechecks/client"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump eslint-plugin-react from 7.24.0 to 7.33.2

Bumps [eslint-plugin-react](https://github.com/jsx-eslint/eslint-plugin-react) from 7.24.0 to 7.33.2.
- [Release notes](https://github.com/jsx-eslint/eslint-plugin-react/releases)
- [Changelog](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/CHANGELOG.md)
- [Commits](https://github.com/jsx-eslint/eslint-plugin-react/compare/v7.24.0...v7.33.2)

---
updated-dependencies:
- dependency-name: eslint-plugin-react
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

OptimismBot committed:
Merge pull request #6977 from ethereum-optimism/refcell/nit-fix-game-dir-comments

chore(op-challenger): Fix Disk Manager Comments

mergify[bot] committed:
Merge branch 'develop' into dependabot/npm_and_yarn/codechecks/client-0.1.12

ajsutton committed:
op-challenger: Use errors.Join

OptimismBot committed:
Merge pull request #6985 from ethereum-optimism/dependabot/npm_and_yarn/codechecks/client-0.1.12

build(deps): bump @codechecks/client from 0.1.11 to 0.1.12

mergify[bot] committed:
Merge branch 'develop' into dependabot/npm_and_yarn/markdownlint-0.30.0

dependabot[bot] committed:
build(deps-dev): bump @babel/eslint-parser from 7.18.2 to 7.22.10

Bumps [@babel/eslint-parser](https://github.com/babel/babel/tree/HEAD/eslint/babel-eslint-parser) from 7.18.2 to 7.22.10.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.22.10/eslint/babel-eslint-parser)

---
updated-dependencies:
- dependency-name: "@babel/eslint-parser"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

OptimismBot committed:
Merge pull request #6984 from ethereum-optimism/dependabot/npm_and_yarn/markdownlint-0.30.0

build(deps-dev): bump markdownlint from 0.24.0 to 0.30.0

mergify[bot] committed:
Merge branch 'develop' into aj/errors-join

OptimismBot committed:
Merge pull request #6990 from ethereum-optimism/aj/errors-join

op-challenger: Use errors.Join

mergify[bot] committed:
Merge branch 'develop' into feat/tx-bundle-builder

OptimismBot committed:
Merge pull request #6898 from ethereum-optimism/feat/tx-bundle-builder

feat: tx bundle builder

mergify[bot] committed:
Merge branch 'develop' into dependabot/npm_and_yarn/eslint-plugin-react-7.33.2

tynes committed:
contracts-bedrock: fix getting started json

Closes https://github.com/ethereum-optimism/optimism/issues/6971

Fixes an issue introduced in https://github.com/ethereum-optimism/optimism/commit/0ee1c09c1399530e0c81f7bf89ef31f455344e4f

Commit:
chore(indexer): Rename l1_tokens and l2_tokens

OptimismBot committed:
Merge pull request #6987 from ethereum-optimism/dependabot/npm_and_yarn/eslint-plugin-react-7.33.2

build(deps-dev): bump eslint-plugin-react from 7.24.0 to 7.33.2

mergify[bot] committed:
Merge branch 'develop' into dependabot/npm_and_yarn/babel/eslint-parser-7.22.10

OptimismBot committed:
Merge pull request #6988 from ethereum-optimism/dependabot/npm_and_yarn/babel/eslint-parser-7.22.10

build(deps-dev): bump @babel/eslint-parser from 7.18.2 to 7.22.10

mergify[bot] committed:
Merge branch 'develop' into refcell/progress-games-on-block

OptimismBot committed:
Merge pull request #6974 from ethereum-optimism/refcell/progress-games-on-block

feat(op-challenger): Progress Games on New L1 Block

mergify[bot] committed:
Merge branch 'develop' into fix/getting-started

OptimismBot committed:
Merge pull request #6991 from ethereum-optimism/fix/getting-started

contracts-bedrock: fix getting started json

Commit:
feat(indexer): add indexer all command

Commit:
feat(indexer): update Dockerfile to run indexer all

mergify[bot] committed:
Merge branch 'develop' into 08-23-chore_indexer_Rename_l1_tokens_and_l2_tokens

OptimismBot committed:
Merge pull request #6992 from ethereum-optimism/08-23-chore_indexer_Rename_l1_tokens_and_l2_tokens

chore(indexer): Rename (l1/l2)_tokens table

maurelian committed:
fix(ctb): Remove unused import from Deploy.s.sol

OptimismBot committed:
Merge pull request #6997 from ethereum-optimism/jm/rm-a-line

fix(ctb): Remove unused import from Deploy.s.sol

ajsutton committed:
op-e2e: Wait for game data to be deleted

Fixes a race condition which could cause flaky e2e tests.

OptimismBot committed:
Merge pull request #6999 from ethereum-optimism/aj/game-data-race

op-e2e: Wait for game data to be deleted

mergify[bot] committed:
Merge branch 'develop' into 08-23-feat_indexer_add_indexer_all_command

OptimismBot committed:
Merge pull request #6994 from ethereum-optimism/08-23-feat_indexer_add_indexer_all_command

feat(indexer): add indexer all command

mergify[bot] committed:
Merge branch 'develop' into 08-23-feat_indexer_update_Dockerfile_to_run_indexer_all

OptimismBot committed:
Merge pull request #6995 from ethereum-optimism/08-23-feat_indexer_update_Dockerfile_to_run_indexer_all

fix(indexer): update Dockerfile to run indexer all

refcell committed:
Deletes the redundant fault dispute game caller component.

OptimismBot committed:
Merge pull request #7003 from ethereum-optimism/refcell/delete-caller

fix(op-challenger): Combine Fault Caller and Loader Components

refcell committed:
Isolates the abi tests to a responder package.

OptimismBot committed:
Merge pull request #7004 from ethereum-optimism/refcell/abi-test-isolation

fix(op-challenger): Isolated Responder ABI Tests

kahuang committed:
Add ability to restrict the networks that op-node discovers peers on

ajsutton committed:
op-program: Always close both sides of a file channel

Ensures the write channel is closed even if the read channel Close returns an error.

mergify[bot] committed:
Merge branch 'develop' into aj/close-all

OptimismBot committed:
Merge pull request #7013 from ethereum-optimism/aj/close-all

op-program: Always close both sides of a file channel

ajsutton committed:
op-challenger: Define worker for parallel scheduling

ajsutton committed:
op-challenger: Simplify worker to a single function.

Commit:
[indexer.configurable_polling] Configurable Polling Variables for Indexer

Commit:
[indexer.configurable_polling] Updated indexer.toml

Commit:
[indexer.configurable_polling] Updated README.md

ajsutton committed:
op-challenger: Implement coordinator

ajsutton committed:
op-challenger: Implement game scheduler

hamdiallam committed:
indexer routine management

hamdiallam committed:
ensure w.Add is called in the same routine as as w.Done

hamdiallam committed:
api logs

hamdiallam committed:
root-level application context. runAll goroutine management

hamdiallam committed:
nit

ajsutton committed:
op-challenger: Play games in parallel

hamdiallam committed:
ci typo

ajsutton committed:
op-challenger: Fix test

Commit:
[indexer.configurable_polling] Updated ChainConfig in e2e/setup.go

mergify[bot] committed:
Merge branch 'develop' into aj/challenger-parallel

OptimismBot committed:
Merge pull request #7017 from ethereum-optimism/aj/challenger-parallel

op-challenger: Play games in parallel

mergify[bot] committed:
Merge branch 'develop' into indexer.configurable_polling

OptimismBot committed:
Merge pull request #7015 from epociask/indexer.configurable_polling

Configurable Polling Variables for Indexer

ajsutton committed:
op-challenger: Add CLI option for max concurrency

Commit:
Add note about arbitrary message sending from OptimismPortal

ajsutton committed:
op-challenger: Use NewConfig helper in e2e tests

OptimismBot committed:
Merge pull request #7028 from mdehoog/withdrawals-note

Add note about arbitrary message sending from OptimismPortal

ajsutton committed:
op-challenger: Separate multigame and single game code into different packages

OptimismBot committed:
Merge pull request #7027 from ethereum-optimism/aj/parallel-cli

op-challenger: Add CLI option for max concurrency

ajsutton committed:
op-challenger: Extract more flexible script to create a new game

ajsutton committed:
op-challenger: Add script to perform moves.

Update visualize script to use cast to list claims so it works with cannon games.

hamdiallam committed:
ETL metrics

mergify[bot] committed:
Merge branch 'develop' into indexer.goroutines

OptimismBot committed:
Merge pull request #7016 from ethereum-optimism/indexer.goroutines

fix(indexer): application-level context. better go-routine management

jyellick committed:
ci: Convert go-e2e-tests CI to make targets

This change makes it much easier to make build changes to the op-e2e
without having to reflect the changes from the Makefile into the CI
definition.  It also makes easier for developers to mirror the execution
of what's going on in CI locally.

mergify[bot] committed:
Merge branch 'develop' into aj/challenger-packages

OptimismBot committed:
Merge pull request #7030 from ethereum-optimism/aj/challenger-packages

op-challenger: Separate multigame and single game code into different…

Commit:
chore(indexer): Use middleware for healthz

OptimismBot committed:
Merge pull request #7037 from ethereum-optimism/08-28-chore_indexer_Use_middleware_for_healthz

chore(indexer): Use middleware for healthz

Inphi committed:
cannon: Prevent deadlock on pre-image reads

Reading a FileChannel that does not have a writer blocks
indefinitely. This occurs in cannon whenever the pre-image oracle
server shuts down prematurely. We add a new io.ReadWriter that can
detect when a FileChannel would otherwise block indefinitely.

hamdiallam committed:
indexer metrics server

hamdiallam committed:
fix unit test

hamdiallam committed:
fix unit tests

ajsutton committed:
op-challenger: Tidy up scripts based on review feedback.

hamdiallam committed:
nit

mergify[bot] committed:
Merge branch 'develop' into indexer.metrics.server

OptimismBot committed:
Merge pull request #7041 from ethereum-optimism/indexer.metrics.server

feat(indexer): introduce metrics server for the indexer process

hamdiallam committed:
clear headers when err == nil

OptimismBot committed:
Merge pull request #7042 from ethereum-optimism/indexer.etl.metrics

feat(indexer): wire up ETL metrics

ajsutton committed:
op-challenger: Use gzip encoded proofs and snapshots

OptimismBot committed:
Merge pull request #7033 from bobanetwork/ci-move-e2e-to-make

ci: Convert go-e2e-tests CI to make targets

ajsutton committed:
op-challenger: Detect already resolved games at startup

Avoids regenerating the cannon traces when performing an initial move at startup if the game is already resolved.

ajsutton committed:
op-challenger: Increase fault game duration on devnet so e2e tests have enough time.

NetWalker108 committed:
Update rollup-node-p2p.md

Typo fix

ajsutton committed:
op-challenger: Include the op-program host binary in docker image.

Avoids the need to manually specify the op-program binary to run as the cannon pre-image server.

OptimismBot committed:
Merge pull request #7045 from NetWalker108/NetWalker108-patch-1

Update rollup-node-p2p.md

mergify[bot] committed:
Merge branch 'develop' into aj/challenger-resolved-game

OptimismBot committed:
Merge pull request #7044 from ethereum-optimism/aj/challenger-resolved-game

op-challenger: Detect already resolved games at startup

mergify[bot] committed:
Merge branch 'develop' into aj/challenger-server-in-docker

OptimismBot committed:
Merge pull request #7046 from ethereum-optimism/aj/challenger-server-in-docker

op-challenger: Include the op-program host binary in docker image.

kahuang committed:
Merge branch 'ethereum-optimism:develop' into netrestrict

OptimismBot committed:
Merge pull request #7034 from conduitxyz/netrestrict

Add ability to restrict the networks that op-node discovers peers on

jyellick committed:
Add cannon-prestate as prereq to devnet-up

The devnet-up ultimately requires the cannont-prestate target to have
already executed, or the script errors out.  Consequently, for someone
checking out the repo if they simply do a:

```
make && make devnet-up
```

Things still fail.  This change simply mirrors a similar check added to
the e2e Makefile.

hamdiallam committed:
e2e setup lifecycle

hamdiallam committed:
client metrics

hamdiallam committed:
subsystem over labels

hamdiallam committed:
nits

hamdiallam committed:
fixes

hamdiallam committed:
test fix

hamdiallam committed:
fix test

hamdiallam committed:
fix cursor

hamdiallam committed:
setup the right next cursor

mergify[bot] committed:
Merge branch 'develop' into aj/challenger-gzip

OptimismBot committed:
Merge pull request #7043 from ethereum-optimism/aj/challenger-gzip

op-challenger: Use gzip encoded proofs and snapshots

Inphi committed:
remove dup waitErr chan read

Inphi committed:
readd sleep before interrupt

mergify[bot] committed:
Merge branch 'develop' into indexer.e2etest.lifecycle

OptimismBot committed:
Merge pull request #7050 from ethereum-optimism/indexer.e2etest.lifecycle

fix(indexer): e2e test lifecycle

OptimismBot committed:
Merge pull request #7047 from ethereum-optimism/inphi/cannon-waiter

cannon: Prevent deadlock on pre-image reads

ajsutton committed:
Merge pull request #7022 from conduitxyz/txmgr-reuse

Allow passing Config and Value to TxMgr

ajsutton committed:
op-chain-ops: Remove old TODO

ajsutton committed:
ci: Require cannon-go-lint-and-test to pass before merging

mergify[bot] committed:
Merge branch 'develop' into engine-api-utils

OptimismBot committed:
Merge pull request #6952 from ethereum-optimism/engine-api-utils

op-wheel: add engine-api utils

mergify[bot] committed:
Merge branch 'develop' into require-cannon-tests

hamdiallam committed:
cursor e2e tests

hamdiallam committed:
fix

hamdiallam committed:
unused var

OptimismBot committed:
Merge pull request #7056 from ethereum-optimism/require-cannon-tests

ci: Require cannon-go-lint-and-test to pass before merging

hamdiallam committed:
Merge pull request #7052 from ethereum-optimism/indexer.etl.subsystems

fix(indexer): Use subsystem rather than label to differentiate between l1/l2

OptimismBot committed:
Merge pull request #7053 from ethereum-optimism/indexer.rpc.metrics

feat(indexer): wire up rpc client metrics

hamdiallam committed:
nits

ajsutton committed:
op-program: Support prefetching l2-transactions.

OptimismBot committed:
Merge pull request #7057 from ethereum-optimism/aj/l2-transactions

op-program: Support prefetching l2-transactions

hamdiallam committed:
run bridge processor after completed L1 intervals

ajsutton committed:
op-program: Use atomic write pattern for KV store

Ensures that pre-image files are always completely written before being moved into place.

hamdiallam committed:
typo

ajsutton committed:
op-challenger: Make cannon processing info logging configurable

Reduce default frequency to avoid spamming the logs too much.

Commit:
[epociask/indexer.architecture] Indexer Architectural Docs

OptimismBot committed:
Merge pull request #7049 from bobanetwork/build-add-cannon-prereq-to-devnet

Add cannon-prestate as prereq to devnet-up

mergify[bot] committed:
Merge branch 'develop' into epociask/indexer.architecture

OptimismBot committed:
Merge pull request #7063 from epociask/epociask/indexer.architecture

[epociask/indexer.architecture] Indexer Architectural Docs

jyellick committed:
op-e2e: allow external processes to skip tests

Certain existing e2e tests are not adaptable to external binary testing.
Although it's better for tests to be able to execute against both an
in-process geth or an extra-process arbitrary ethereum client, the
current state of the tests does not allow this in all cases.

This change allows for external binaries to specify a set of test names
to be skipped, including a skip message for why they are skipped.  The
set is specified as a JSON map in a new optional file `test_parms.json`
which can be created in the external adapter folder.

Prior to this change, the external process was indicated only by a path
to a `shim` binary.  This change now instead asks that the external test
be specified by directory, wherein a `shim` binary must exist, and this
new file `test_params.json` may exist.  Other approaches are of course
possible, such as allowing the JSON file to be specified via a command
flag, but it seemed better to simply establish a convention for expected
paths for these external tests to satisfy.

There actually is a single test which already has a `t.Skip()` based on
the presence of the external L2 flag.  I'd though these were all
removed in the last PR, but this one actually is required for an
external op-geth.  So, this skip has been converted to the new method
and acts as a nice test that it is working.

hamdiallam committed:
Merge pull request #7054 from ethereum-optimism/indexer.transfer.cursoring

fix(indexer): cursoring options when scanning bridge transfers

mergify[bot] committed:
Merge branch 'develop' into aj/atomic-put

hamdiallam committed:
confirmation depth

OptimismBot committed:
Merge pull request #7060 from ethereum-optimism/aj/atomic-put

op-program: Use atomic write pattern for KV store

hamdiallam committed:
toml update and test fix

hamdiallam committed:
fix header_traversal_test

hamdiallam committed:
Merge pull request #7058 from ethereum-optimism/indexer.bridge.interval

fix(indexer): run bridge processor after completed L1ETL intervals

hamdiallam committed:
Merge pull request #7066 from ethereum-optimism/indexer.conf.depth

feat(indexer): configurable confirmation depth when traversing headers

ajsutton committed:
Merge remote-tracking branch 'origin/develop' into aj/remove-todo

ajsutton committed:
op-chain-ops: Detect types using input.Type.T rather than string comparison

OptimismBot committed:
Merge pull request #7055 from ethereum-optimism/aj/remove-todo

op-chain-ops: Remove old TODO

OptimismBot committed:
Merge pull request #7065 from bobanetwork/e2e-externals-specify-test-skips

op-e2e: allow external processes to skip tests

ajsutton committed:
op-challenger: Add documentation for manual testing scripts.

ajsutton committed:
op-challenger: Use the checkpointed block number from the emitted event rather than subtracting one from the block number manually.

mergify[bot] committed:
Merge branch 'develop' into aj/better-type-check

OptimismBot committed:
Merge pull request #7070 from ethereum-optimism/aj/better-type-check

op-chain-ops: Detect types using input.Type.T rather than string comparison

mergify[bot] committed:
Merge branch 'develop' into aj/cannon-info-freq

OptimismBot committed:
Merge pull request #7062 from ethereum-optimism/aj/cannon-info-freq

op-challenger: Make cannon processing info logging configurable

mergify[bot] committed:
Merge branch 'develop' into aj/game-scripts

OptimismBot committed:
Merge pull request #7031 from ethereum-optimism/aj/game-scripts

op-challenger: Make scripts more flexible

jbearer committed:
Merge remote-tracking branch 'upstream/develop' into update/upstream

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/EspressoSystems/op-espresso-integration?pullRequest=57) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/EspressoSystems/op-espresso-integration?pullRequest=57) before we can accept your contribution.<br/>**0** out of **6** committers have signed the CLA.<br/><br/>:x: Ethen Pociask<br/>:x: hamdiallam<br/>:x: OptimismBot<br/>:x: ajsutton<br/>:x: mergify[bot]<br/>:x: jyellick<br/><hr/>**Ethen Pociask** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/EspressoSystems/op-espresso-integration?pullRequest=57) it.</sub>

jbearer committed:
Run e2e tests on xlarge instance

This is the configuration specified for these tests in the
upstream repo. It got lost somewhere in my changes. This may fix
the timeout errors that we are getting.

jbearer committed:
Update op-geth

sveitser commented:
@jbearer Should we merge this?

jbearer commented:
I want to wait for #55 so that if we get conflicts I can fix them here instead of blocking that critical PR.

jbearer committed:
Merge remote-tracking branch 'origin/integration' into update/upstream
